flink-autoscaler/src/main/java/org/apache/flink/autoscaler/topology/JobTopology.java

```diff
@@ -146,13 +145,18 @@ public static JobTopology fromJsonPlan(
         for (JsonNode node : nodes) {
             var vertexId = JobVertexID.fromHexString(node.get("id").asText());
             var inputList = new HashSet<JobVertexID>();
+            var ioMetrics = metrics.get(vertexId);
+            var finished = finishedVertices.contains(vertexId);
             vertexInfo.add(
                     new VertexInfo(
                             vertexId,
                             inputList,
+                            null,
                             node.get("parallelism").asInt(),
                             maxParallelismMap.get(vertexId),
-                            finished.contains(vertexId)));
+                            maxParallelismMap.get(vertexId),
```

Comment:
I think we want to remove the `maxParallelism` parameter and do the max parallelism housekeeping internally.

Reply:
This might stem from Lombok Setters/Getters but I think it makes it less clear that only maxParallelism is mutated and originalMaxParallelisms is immutable.

Reply:
I will add an explicit constructor to clean this up instead of relying on Lombok allArgsConstructor